plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ProgramConverter.kt

```diff
@@ -112,6 +111,20 @@ class ProgramConverter(val session: FirSession, override val config: PluginConfi
         // and then later decide to add a body anyway.  It's not a problem for now, but
         // worth being aware of.
         return methods.getOrPut(signature.name) {
+            val contractVisitor = ContractDescriptionConversionVisitor(this@ProgramConverter, signature)
+            val parameterIndices = (signature.params.indices.toSet() + setOfNotNull(signature.receiver?.let { -1 })).toMutableSet()
+            val nonDuplicableIndices = symbol.resolvedContractDescription?.effects?.mapNotNull { decl ->
+                (decl.effect as? KtCallsEffectDeclaration<*, *>)?.valueParameterReference?.parameterIndex
+            }?.toSet() ?: emptySet()
+
+            val duplicableParameters =
+                (parameterIndices - nonDuplicableIndices).map { contractVisitor.embeddedVarByIndex(it) }.filter { it.type is FunctionTypeEmbedding }
```

Comment:
I think that writing chaining calls with this style can make the code a bit more readable:
```kotlin
  val parameterIndices = (signature.params.indices.toSet() + setOfNotNull(signature.receiver?.let { -1 }))
      .toMutableSet()
  val nonDuplicableIndices = symbol.resolvedContractDescription?.effects
      ?.mapNotNull { decl -> (decl.effect as? KtCallsEffectDeclaration<*, *>)?.valueParameterReference?.parameterIndex }
      ?.toSet()
      ?: emptySet()

  val duplicableParameters = (parameterIndices - nonDuplicableIndices)
      .map { contractVisitor.embeddedVarByIndex(it) }
      .filter { it.type is FunctionTypeEmbedding }
      .map { DuplicableFunction.toFuncApp(listOf(it.toLocalVar())) }
```
Since `processFunction` is becoming really big, maybe it can be the case to refactor these lines in a private function

Reply:
There's actually a refactoring coming up that will split this up: https://github.com/jesyspa/kotlin/pull/90/commits/b6d85c7b21c4c8f7e2dd010f3d04553ed7b588b1
It's a good point, I'll apply that there—we'll probably want some helper functions, too.